src/installing-deis/configuring-object-storage.md

```diff
@@ -6,20 +6,103 @@ A variety of Deis components rely on an object storage system to do their work.
 - [slugbuilder](https://github.com/deis/slugbuilder)
 - [slugrunner](https://github.com/deis/slugrunner)
 - [registry](https://github.com/deis/registry)
+- [database](https://github.com/deis/postgres)
 
-These components are built flexibly, so they can work out of the box with almost any system that is compatible with the [S3 API](http://docs.aws.amazon.com/AmazonS3/latest/API/APIRest.html).
+These components are flexible and can work out of the box with almost any system that is compatible with the [S3 API](http://docs.aws.amazon.com/AmazonS3/latest/API/APIRest.html).
 
 # Minio
 
 Additionally, Deis ships with a [Minio](http://minio.io) [component](https://github.com/deis/minio). This component runs as a Kubernetes service, and the components listed above are configured to automatically look for that service and use it as object storage if it's available.
 
-# Telling Deis What to Use
+# Configuring the Deis Components
 
-The Deis components determine what object storage system to use via environment variables that you set up:
+Every Deis component that relies on object storage relies on the following two inputs for configuration:
 
-- `DEIS_MINIO_SERVICE_HOST` and `DEIS_MINIO_SERVICE_PORT` - The in-cluster Minio service
-- `DEIS_OUTSIDE_STORAGE_HOST` and `DEIS_OUTSIDE_STORAGE_PORT` - The external S3-compatible object storage system
+- One or more environment variables with host and port to describe where the object storage system is
+- One or more files to provide access credentials for the object storage system.
+	- We suggest storing these values in [Kubernetes secrets](http://kubernetes.io/v1.1/docs/user-guide/secrets.html) and mounting them as volumes to each pod
+	- See [the deis-dev chart](https://github.com/deis/charts/tree/master/deis-dev) for examples of using and mounting secrets.
 
-# Limitations
+The subsections herein explain how to configure these two inputs for each applicable component.
 
-The only currently known limitation is that [the Deis registry component](https://github.com/deis/registry) will not automatically look up the minio service, nor will it look for other storage env vars. That fix is being tracked in a [GitHub issue](https://github.com/deis/registry/issues/7) and is planned for our beta release.
+## [deis/builder](https://github.com/deis/builder)
+
+### Environment Variables
+
+The builder looks for the below environment variables to determine where the object storage system is. The builder looks in-order for these variables. If it finds two, the one higher in the list will be used.
```

Comment:
It might be easier to just explicitly state the preferred variable in the case of all of them being present. E.g.,

If it finds two, `DEIS_OUTSIDE_STORAGE` will take precedence and external object storage configuration will be applied.